cms/lib/xblock/upstream_sync.py

```diff
@@ -274,17 +293,38 @@ def decline_sync(downstream: XBlock) -> None:
 
     Does not save `downstream` to the store. That is left up to the caller.
 
-    If the syncing to `downstream` is not supported, raises BadDownstream.
-    If the upstream link is invalid or broken, raises BadUpstream.
-    If `downstream` is not linked at all, does nothing.
+    If `downstream` lacks a valid+supported upstream link, this raises an UpstreamLinkException.
     """
     # Try to load the upstream.
-    upstream_link = UpstreamLink.get_for_block(downstream)  # Can raise BadUpstream or BadUpstream
+    upstream_link = UpstreamLink.get_for_block(downstream)  # Can raise UpstreamLinkException
     if not upstream_link:
         return  # No upstream -> nothing to sync.
     downstream.upstream_version_declined = upstream_link.version_available
 
 
+def sever_upstream_link(downstream: XBlock) -> None:
+    """
+    Given an XBlock that is linked to upstream content, disconnect the link, such that authors are never again prompted
+    to sync upstream updates. Erase all `.upstream*` fields from the downtream block.
+
+    However, before nulling out the `.upstream` field, we copy its value over to `.copied_from_block`. This makes sense,
+    because once a downstream block has been de-linked from source (e.g., a Content Library block), it is no different
+    than if the block had just been copy-pasted in the first place.
+
+    Does not save `downstream` to the store. That is left up to the caller.
+
+    If `downstream` lacks a link, then this raises NoUpstream (though it is reasonable for callers to handle such
+    exception and ignore it, as the end result is the same: `downstream.upstream is None`).
+    """
+    if not downstream.upstream:
+        raise NoUpstream()
+    downstream.copied_from_block = downstream.upstream
+    downstream.upstream = None
+    downstream.upstream_version = None
+    for _, fetched_upstream_field in downstream.get_customizable_fields().items():
+        setattr(downstream, fetched_upstream_field, None)  # Null out upstream_display_name, et al.
+
+
```

Comment:
@ormsbee Note that I've gone an implemented `sever_upstream_link` (and its matching HTTP API, `DELETE /api/contentstore/v2/downstream/{key}`. It's not needed for the MVP, but it was easy enough to do, and it ensures that we don't forget [Braden's copied_from_block suggestion here](https://github.com/openedx/edx-platform/pull/34925#discussion_r1797412093).